unit-tests/src/test/scala/scala/scalanative/posix/sys/WaitTest.scala

```diff
@@ -0,0 +1,58 @@
+package scala.scalanative.posix.sys
+
+import scala.scalanative.posix.sys.wait._
+import scala.scalanative.posix.unistd.vfork
+import scala.scalanative.unsafe._
+
+import org.junit.Test
+import org.junit.Assert._
+
+class WaitTest {
+  @Test def waitWaitsChildExit(): Unit = {
+    val statLoc = stackalloc[CInt](1)
+
+    def blackHole(a: Any): Unit = ()
+
+    /** these are just here to test the bindings compile and link **/
+    blackHole(WCONTINUED)
+    blackHole(WNOHANG)
+    blackHole(WUNTRACED)
+
+    WIFSIGNALED(!statLoc)
+    WTERMSIG(!statLoc)
+    WIFSTOPPED(!statLoc)
+    WSTOPSIG(!statLoc)
+    WIFCONTINUED(!statLoc)
+
+    //done
+
+    val pid = vfork()
+    if (pid == 0) {
```

Comment:
Not a requirement but a personal style.

I have been taught, and have taught my students, to put the "short'
branches before the long branches.  I would probably order the 
cases  < 0, 0, >0, where >0 is the long branch.  This helps a reader
consider the short cases first, rather than have a dangling short branch.
Given the semantics & usual usage of `vfork()` there is a reason for doing
0,  < 0, > 0 also (i.e. being in child(?) will happen more often than errors.)
Of course, the reason for putting errors at the bottom is that they are least
likely.   

I think checking errors first comes from Fortran practice and the cost of
branches & jumps.  Old tapes...

Reply:
done